packages/expo-application/ios/EXApplication/EXApplication.m

```diff
@@ -3,6 +3,8 @@
 #import <EXApplication/EXApplication.h>
 #import <UIKit/UIKit.h>
 
+NSString * const kEXApplicationInstallationIdKey = @"kEXApplicationInstallationIdKey";
```

Comment:
Thanks to `EXScopedApplication` and `ScopedApplicationModule` this doesn't matter in ExpoKit applications — `Kernel`/`ExponentSharedPreferences` installationID is used in `expo-application` there. This key will only be used in new `expo-application` applications. Or maybe your concern applies to applications ejected to bare (that we would like to use the same UUID there)?

Reply:
My concerns are that existing standalone apps need to read the existing data (we cannot generate a new UUID) and we should deprecate Constants.installationId and not provide a replacement (if developers want to re-implement this in user space they can do that, but we don't want to encourage it).

Reply:
But we still need to have a UUID to be able to register the push token in the Expo server, right? Or would you want to deprecate this argument and identify devices by tokens?

If I understood the matter correctly the PR at current state should be ok. 🙂

Reply:
> But we still need to have a UUID to be able to register the push token in the Expo server, right?

We need to have a unique ID but it does not necessarily need to be a UUID. 

>  Or would you want to deprecate this argument and identify devices by tokens?

We cannot identify devices by push tokens for two reasons. Push tokens can change at any time. We also don't want to store the previous push token because if a user gets a new phone and restores it from backup, we'll see two devices that report having had the same previous push token.

Reply:
> We need to have a unique ID but it does not necessarily need to be a UUID.

At the moment server doesn't accept anything that isn't a UUIDv4, I experimented with using `Secure.ANDROID_ID` before and it failed (see [universe schema](https://github.com/expo/universe/blob/b60d7d4734940bfb234a9c545234a87aa2f3d8a7/server/www/src/apiv2/PushRouter.js#L31-L33)).

> Push tokens can change at any time.

That's why I would have imagined invalidating tokens that return `unregistered` from Push Notification servers in our storage.

> if a user gets a new phone and restores it from backup, we'll see two devices that report having had the same previous push token

If we identified devices by tokens (there wouldn't be a notion of `device` on server) in this situation there would be two tokens (after all a token must identify a single device, doesn't it?)

---

All in all I think making any of these changes would be non-trivial and if we ever wanted to revolutionize the API this would require giving it a good thinking. (+ my comments run around an idea of _removing UUID_ and it may not be a good idea – I was just curious about whether it would be doable.)

Reply:
> That's why I would have imagined invalidating tokens that return unregistered from Push Notification servers in our storage.

We can only invalidate a token when a push notification is sent. If the developer doesn't send push notifications but the app changes its token, we don't get notified that the old token is invalid.

> If we identified devices by tokens (there wouldn't be a notion of device on server) in this situation there would be two tokens (after all a token must identify a single device, doesn't it?)

Each device would have a different new token and have the same old token. The server would need to handle this "fork".